File: oqueue-flutter/lib/domain.dart
class QueueInfo {
  final String id;
  final String name;
  final String description;
  final String organizerId;
  final String createdAt;
  final String existsBefore;

  const QueueInfo(this.id, this.name, this.description, this.organizerId, this.createdAt, this.existsBefore);
}

class MemberInfo {
  final String id;
  final int order;
  final bool hasPriority;
  final bool isHeld;
  final String joinedAt;

  const MemberInfo(this.id, this.order, this.hasPriority, this.isHeld, this.joinedAt);
}

class FullMemberInfo {
  final String id;
  final String name;
  final int order;
  final bool hasPriority;
  final bool isHeld;
  final String joinedAt;

  const FullMemberInfo({
    required this.id,
    required this.name,
    required this.order,
    required this.hasPriority,
    required this.isHeld,
    required this.joinedAt});
}

class UserInfo {
  final String id;
  final String name;

  const UserInfo({required this.id, required this.name});
}

File: oqueue-server/src/domain.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use uuid::Uuid;

pub type Id = Uuid;

// ---------
// UserId
// ---------

#[derive(Clone, Hash, Debug)]
pub struct UserId(pub Id);

// ---------
// QueueId
// ---------

#[derive(Clone, Hash, Debug)]
pub struct QueueId(pub Id);

// --------
// User
// --------

#[derive(Clone, Hash, Debug)]
pub struct User {
    id: UserId,
    name: String,
    email: String,
    pwhash: String,
}

// -------
// Other Structures
// -------

#[derive(Clone, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
pub struct UserInfo {
    pub id: Uuid,
    pub name: String,
}

#[derive(Clone, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
pub struct MemberInfo {
    pub id: Uuid,
    pub order: i32,
    pub has_priority: bool,
    pub is_held: bool,
    pub joined_at: NaiveDateTime,
}

#[derive(Clone, Eq, PartialEq, Hash, Debug, Serialize, Deserialize)]
pub struct QueueInfo {
    pub id: Uuid,
    pub name: String,
    pub description: String,
    pub organizer_id: Uuid,
    pub created_at: NaiveDateTime,
    pub exists_before: NaiveDateTime,
}
